sdk/azidentity/managed_identity_client.go

```diff
@@ -60,7 +60,10 @@ const (
 )
 
 type managedIdentityClient struct {
-	azClient  *azcore.Client
+	azClient *azcore.Client
+	// chained indicates whether the client is part of a credential chain. If true, the client will return
+	// a credentialUnavailableError instead of an AuthenticationFailedError for an unexpected IMDS response.
+	chained   bool
```

Comment:
I must say that changing the error type based on the calling context makes me a bit nervous, but maybe I'm overthinking it. If there isn't a cleaner way to do this, I can live with it though (or if we already have this pattern elsewhere).

This should be moved to after `probeIMDS` though in order to avoid some unnecessary padding in the struct layout (we need to lint for this at some point as I'm sure there are other cases in the codebase).